src/test/java/RecipeTest.java

```diff
@@ -0,0 +1,84 @@
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Arrays;
+
+/** */
+@RunWith(JUnit4.class)
+public final class RecipeTest {
+
+  private static final String NAME = "Grilled Cheese";
+  private static final String DESCRIPTION = "It's literally just melted cheese on bread";
+  private static final List<Step> STEPS = Arrays.asList(
+      new Step("Toast the bread for 2 minutes"),
+      new Step("Melt the cheese"),
+      new Step("Put the cheese in the bread")
+  );
+  private Recipe recipe;
+
+  @Before
+  public void setup() {
+    recipe = new Recipe(NAME, DESCRIPTION, new LinkedList(STEPS));
+  }
+
+  @Test
+  public void appendStep() {
+    List<Step> expectedSteps = new LinkedList<>(STEPS);
+    expectedSteps.add(new Step("butter the bread"));
+
+    Recipe expected = new Recipe(NAME, DESCRIPTION, expectedSteps);
+    recipe.appendStep(new Step("butter the bread"));
+
+    Assert.assertEquals(expected, recipe);
+  }
+
+  @Test
+  public void addStepOutOfBounds() {
+    List<Step> expectedSteps = new LinkedList<>(STEPS);
+    expectedSteps.add(new Step("butter the bread"));
+
+    Recipe expected = new Recipe(NAME, DESCRIPTION, expectedSteps);
+    recipe.addStep(797234, new Step("butter the bread"));
+    Assert.assertEquals(expected, recipe);
+
+    recipe.addStep(-5, new Step("I'm outta bounds!"));
```

Comment:
I don't believe you're returning an error message in your code (per my comments above), so how exactly are you testing this? Maybe I'm missing something, but an explanation would be nice; my understanding now is that you're looking for the print statement "I'm outta bounds!" but won't actually get an assertion error (which is convention for testing, I believe).

Reply:
The test was that adding a step at position <0 adds nothing, and adding a step >steps.size() just appends the step.